ompi/mca/coll/hcoll/coll_hcoll_ops.c

```diff
@@ -136,8 +137,14 @@ int mca_coll_hcoll_allgatherv(const void *sbuf, int scount,
                                              hcoll_module->previous_allgatherv_module);
         return rc;
     }
-    rc = hcoll_collectives.coll_allgatherv((void *)sbuf,scount,stype,rbuf,(int*)rcount,
-                                           (int*)displs,rtype,hcoll_module->hcoll_context);
+    rc = hcoll_collectives.coll_allgatherv((void *)sbuf,
+                                           scount,
+                                           stype,
+                                           rbuf,
+                                           (void *)ompi_count_array_ptr(rcount),
+                                           (void *)ompi_disp_array_ptr(displs),
```

Comment:
Are these casts needed?

Reply:
@jtronge Ping

Reply:
without these cases the compiler (gcc) squawks some:
```
coll_hcoll_ops.c: In function ‘mca_coll_hcoll_allgatherv’:
coll_hcoll_ops.c:144:44: warning: passing argument 5 of ‘hcoll_collectives.coll_allgatherv’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
                                            ompi_count_array_ptr(rcount),
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
coll_hcoll_ops.c:144:44: note: expected ‘int *’ but argument is of type ‘const void *’
coll_hcoll_ops.c:145:44: warning: passing argument 6 of ‘hcoll_collectives.coll_allgatherv’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
                                            ompi_disp_array_ptr(displs),
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~
coll_hcoll_ops.c:145:44: note: expected ‘int *’ but argument is of type ‘const void *’
In file included from coll_hcoll_ops.c:17:
```

Reply:
Ahh, I missed that the casts were there in the original code too. Thanks!